### PR TITLE
"What's new?" page

### PR DESCRIPTION
**Pull request by axemonkey**
## What

Implement a new page at `/whats-new` which pulls in [markdown from the platform guidance repo](https://raw.githubusercontent.com/moj-analytical-services/platform_user_guidance/master/changelog/whats_new.md) and displays it. Also display a notification icon in the header bar when there are unseen notifications.

## How to review

1. Check out and see that there is a grey '?' icon in the top right of the header bar on, for example, the homepage. This indicates there is something new on the "What's new?" page that you haven't seen
2. Click the icon to visit the "What's new?" page – the icon will disappear, as you have now seen whatever it is that is new, and a cookie is created containing a hash of the markdown content
3. Go back to the homepage or to any other page, the icon should not display. Next time the what's new page is changed, the icon will reappear until the page is visited again and the cookie is updated

### CODE REVIEW

**Comment by axemonkey** [comment]
Gah. Forgot to use `url_for()`...

File: app/assets/javascripts/modules/cookies.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  },
+
+  store(name, value, days) {
+    let expires;
+    const dayInMS = 24 * 60 * 60 * 1000;
+
+    if (days) {
+      const date = new Date();
+      date.setTime(date.getTime() + (days * dayInMS));
+      expires = `; expires=${date.toGMTString()}`;
```
[the remark this answers is not in the document]
Comment (axemonkey) [reply app/assets/javascripts/modules/cookies.js]: All of the above was copypasta that I meant to go back and revisit, but forgot. Sorry for making you go through it all Andy. 😞

Will address.

File: app/assets/javascripts/modules/whats-next-notification.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+moj.Modules.whatsNextNotification = {
```
[the remark this answers is not in the document]
Comment (axemonkey) [reply app/assets/javascripts/modules/whats-next-notification.js]: That's just total brainfart. Thanks.

File: app/assets/javascripts/modules/cookies.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+moj.Modules.cookies = {
+  read(name) {
+    const nameEQ = `${encodeURIComponent(name)}=`;
+    const ca = document.cookie.split(';');
+
+    for (let i = 0; i < ca.length; i += 1) {
+      let c = ca[i];
```
[the remark this answers is not in the document]
Comment (axemonkey) [reply app/assets/javascripts/modules/cookies.js]: Did not know about `for (let item of array) {}` – thx! :)

File: app/assets/javascripts/modules/cookies.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      while (c.charAt(0) === ' ') {
+        c = c.substring(1, c.length);
+      }
+      if (c.indexOf(nameEQ) === 0) {
+        return decodeURIComponent(c.substring(nameEQ.length, c.length));
+      }
+    }
+    return null;
```
[the remark this answers is not in the document]
Comment (axemonkey) [reply app/assets/javascripts/modules/cookies.js]: eslint disagrees
```
2:3   error  Expected to return a value at the end of method 'read'
consistent-return
```

File: app/assets/javascripts/modules/cookies.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (axemonkey) [reply app/assets/javascripts/modules/cookies.js]: `cookies.forEach()` FTW.

File: app/assets/javascripts/modules/cookies.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+moj.Modules.cookies = {
+  read(name) {
+    const nameEQ = `${encodeURIComponent(name)}=`;
+    const ca = document.cookie.split(';');
+
+    for (let i = 0; i < ca.length; i += 1) {
+      let c = ca[i];
+      while (c.charAt(0) === ' ') {
+        c = c.substring(1, c.length);
+      }
```
[the remark this answers is not in the document]
Comment (axemonkey) [reply app/assets/javascripts/modules/cookies.js]: 👍

File: app/config.js [hunk 6]
```diff
@@ -66,6 +66,10 @@ config.cookie = {
   secret: process.env.COOKIE_SECRET || 'shh-its-a-secret',
 };
 
+config.marked = {
+  gfm: true,
```
[the remark this answers is not in the document]
Comment (axemonkey) [reply app/config.js]: 👍

File: app/assets/javascripts/modules/cookies.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+moj.Modules.cookies = {
+  read(name) {
+    const nameEQ = `${encodeURIComponent(name)}=`;
+    const cookies = document.cookie.split(/\s*;\s*/);
+    let cookieValue;
+
+    cookies.forEach((cookie) => {
+      if (cookie.indexOf(nameEQ) === 0) {
+        cookieValue = decodeURIComponent(cookie.substring(nameEQ.length, cookie.length));
```
Comment (andyhd) [inline app/assets/javascripts/modules/cookies.js]: One small thing - this continues looping through the cookies after it has found the one you want. You could use `cookies.find(cookie => cookie.startsWith(nameEQ))` instead.

Comment (axemonkey) [reply app/assets/javascripts/modules/cookies.js]: Ooh, like it. 👍

File: app/assets/javascripts/modules/cookies.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (andyhd) [reply app/assets/javascripts/modules/cookies.js]: bah

**Pull request merged**